.semaphore/semaphore.yml

```diff
@@ -0,0 +1,123 @@
+---
+version: v1.0
+name: UPM
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+
+global_job_config:
+  prologue:
+    commands:
+      - checkout
+      - git switch --detach
+      - export PATH="/tmp:$PATH"
+
+blocks:
+  - name: setup
+    dependencies: []
+    task:
+      jobs:
+        - name: cache nix installer
+          commands:
+            - >-
+              cache has_key nix-installer || (
+              curl
+              --proto '=https'
+              -o /tmp/nix-installer
+              --tlsv1.2 -sSf -L
+              https://github.com/DeterminateSystems/nix-installer/releases/download/v0.7.0/nix-installer-x86_64-linux
+              && chmod +x /tmp/nix-installer
+              && cache store nix-installer /tmp/nix-installer
+              )
+            # TODO: cache the dependencies of the nix package
+            # this might be possible based on exploratory:
+            # ```
+            # $ DERIVATION="$(nix path-info --derivation .)"
+            # $ declare -p DERIVATION
+            # $ nix-store --query --references "${DERIVATION}"
+            # ```
+            # The above spits out the deps of the derivation
+            # just have to filter out the `-source`
+        - name: cache golint-ci
+          commands:
+            - >-
+              cache has_key golangci-lint || (
+              mkdir -p bin &&
+              curl
+              -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh
+              | sh -s -- -b /tmp v1.52.2
+              && cache store golangci-lint /tmp/golangci-lint
+              )
+
+  - name: build
+    dependencies: [ setup ]
+    task:
+      prologue:
+        commands:
+          - cache restore nix-installer
+          - sudo /tmp/nix-installer install linux --no-confirm --logger compact
+          - . /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
+      jobs:
+        - name: nix
+          commands:
+            - cache restore nix-store
+            - nix build
+            - ./result/bin/upm --version
+            - cache store nix-store /nix
+        # TODO: get this working again
+        # - name: docker
+        #   commands:
+        #     - make light full
```

Comment:
some package we depend on in the docker images doesn't have the right version when building the image, i figured it wasn't important enough to keep diagnosing

Reply:
yeah let's skip the docker builds